add wifi-aware.md guide page (#2088)
Samsung/tizen-docs: docs/application/native/guides/connectivity/wifi-aware.md

Wi-Fi Aware networking operates by establishing clusters. A Wi-Fi Aware device can join an existing cluster or create a new cluster if that device is the first in the area. Once a cluster is established, devices can start service discvoery and messaging using publish, subscribe and send message APIs. After service discovery data path can be established by devices to send and receive data over network layer.
## Features

The main features of the Wi-Fi Aware API include the following:

- Initialization and deinitialization of Wi-Fi Aware functionalities
  You can initialize Wi-Fi Aware functionalities using the `wifi_aware_initialize()` API and deinitialize them using the `wifi_aware_deinitialize()` API. These APIs allow you to initialize and deinitialize the Wi-Fi Aware module in your application.

- Enabling and disabling Wi-Fi Aware functionalities.
  You can [enable](#enabling-wi-fi-aware) or [disable](#disabling-wi-fi-aware) Wi-Fi Aware functionalities.

> - Creating and destroying Wi-Fi Aware sessions
>   You can [create a new Wi-Fi Aware session](#session-creation) and [destroy an existing session](#disabling-wi-fi-aware)

Reviewer: ```suggestion
- Creating and destroying Wi-Fi Aware sessions.
  You can [create a new Wi-Fi Aware session](#session-creation) and [destroy an existing session](#disabling-wi-fi-aware).
```

- Starting and Stopping Wi-Fi Aware sessions.

>   Based on session type you can [start publish session](#publish-service) or [subscribe session](#subscribe-service)
>   You can [stop an active Wi-Fi Aware session](#stop-service-discovery).

Reviewer: ```suggestion
  Based on session type you can [start publish session](#publish-service) or [subscribe session](#subscribe-service).
  You can [stop an active Wi-Fi Aware session](#stop-service-discovery).
```

> - Updating existing publish session
>   You can update an existing publish session using the `wifi_aware_session_update_publish()` API. This allows you to modify the parameters of an ongoing publish service session.

Reviewer: ```suggestion
- Updating existing publish session.
  You can update an existing publish session using the `wifi_aware_session_update_publish()` API. This allows you to modify the parameters of an ongoing publish service session.
```

> - Updating existing subscribe session
>   You can update an existing subscription using the `wifi_aware_session_update_subscribe()` API. This allows you to modify the parameters of an ongoing subscribe service session.

Reviewer: ```suggestion
- Updating existing subscribe session.
  You can update an existing subscription using the `wifi_aware_session_update_subscribe()` API. This allows you to modify the parameters of an ongoing subscribe service session.
```

> - Sending messages to peers
>   You can [send messages to peers](#send-message).

Reviewer: ```suggestion
- Sending messages to peers.
  You can [send messages to peers](#send-message).
```

> - Setting up data paths between peers
>   You can [set up data path](#data-path-setup) between peers.

Reviewer: ```suggestion
- Setting up data paths between peers.
  You can [set up data path](#data-path-setup) between peers.
```

To enable a Wi-Fi Aware local device:

1. Define callbacks for the Wi-Fi Aware enable/disable events:

   Define the `wifi_aware_enabled_cb()` and `wifi_aware_disabled_cb()` callback function, which are invoked whenever a Wi-Fi Aware local device enables or disables. In the following example, an information message is printed in the console.

   ```
	static void __wifi_aware_enabled_cb(wifi_aware_error_e error, void *user_data)
	{
		if (error == WIFI_AWARE_ERROR_NONE)
			printf("Wi-Fi Aware is enabled\n");
		else
			printf("Wi-Fi Aware is not enabled. Error: %d\n", error);
	}

	static void __wifi_aware_disabled_cb(wifi_aware_error_e error, void *user_data)
	{
		if (error == WIFI_AWARE_ERROR_NONE)
			printf("Wi-Fi Aware is disabled\n");
		else
			printf("Wi-Fi Aware is not disabled. Error: %d\n", error);
	}
   ```

2. Enable the Wi-Fi Aware local device:

   After the `wifi_aware_enable()` function is completed, the `wifi_aware_enabled_cb()` callback is invoked.

   ```
	error_code = wifi_aware_enable(__wifi_aware_enabled_cb, NULL);
   ```

<a name="session_creation"></a>
## Session Creation

> Wi-Fi aware session will be required to perform Wi-Fi aware operations like service discovery, data path creation, etc. on the device.

Reviewer: ```suggestion
Wi-Fi Aware session will be required to perform Wi-Fi Aware operations such as, service discovery, data path creation, etc., on the device.
```
